File: backend/src/main/java/org/example/backend/model/Expense.java
package org.example.backend.model;

import lombok.With;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.time.LocalDate;

@With
public record Expense(
        @MongoId
        String id,
        String category,
        String vendor,
        double amount,
        boolean isCashPayment,
        String description,
        LocalDate date
) {
}

File: frontend/src/model/Expense.ts
export type Expense = {
    id: string,
    category: string,
    vendor: string,
    amount: number,
    isCashPayment: boolean,
    description: string,
    date: string
    paymentFrequency: "once" | "weekly" | "monthly" | "quarterly" | "biannual" | "yearly"
}